Sources/SpeziQuestionnaire/QuestionnaireConstraint.swift

```diff
@@ -0,0 +1,18 @@
+//
+// This source file is part of the Stanford Spezi open-source project
+//
+// SPDX-FileCopyrightText: 2022 Stanford University and the project authors (see CONTRIBUTORS.md)
+//
+// SPDX-License-Identifier: MIT
+//
+
+import ModelsR4
+import Spezi
+
+
+/// A Constraint which all `Standard` instances must conform to when using the Spezi Questionnaire module.
+public protocol QuestionnaireConstraint: Standard {
+    /// Adds a new `QuestionnaireResponse` to the `Standard` conforming to `QuestionnaireConstraint`.
+    /// - Parameter response: The `QuestionnaireResponse` that should be added.
+    func add(_ response: QuestionnaireResponse) async
```

Comment:
In sync to the HealthKit PR, I would suggest that we slightly adjust the function signature here to make it easier to multiplex between the different add methods we might need to implement in a `Standard` conforming to the different constraints:
```suggestion
    func add(response: QuestionnaireResponse) async
```